servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/HttpMessageDiscardWatchdogClientFilterTest.java

```diff
@@ -0,0 +1,197 @@
+     */
+    @ParameterizedTest(name = "{displayName} [{index}] filterType={0} expectedException={1} transformer={2}")
+    @MethodSource("responseTransformers")
+    void cleansClientResponseMessageBodyIfDiscarded(final FilterType filterType,
+                                                    final @Nullable Class<?> expectedException,
+                                                    ResponseTransformer transformer)
+            throws Exception {
+
+        try (HttpServerContext serverContext = newServerBuilder(SERVER_CTX)
+                .listenStreamingAndAwait((ctx, request, responseFactory) ->
+                        Single.fromSupplier(() -> responseFactory.ok().payloadBody(Publisher.from(ctx.executionContext()
+                                .bufferAllocator().fromUtf8("Hello, World!")))))) {
+            try (StreamingHttpClient client = newClientBuilder(serverContext, CLIENT_CTX)
+                    .appendConnectionFilter(c -> new StreamingHttpConnectionFilter(c) {
+                        @Override
+                        public Single<StreamingHttpResponse> request(final StreamingHttpRequest request) {
+                            if (filterType.equals(FilterType.CONNECTION)) {
+                                return transformer.apply(delegate(), request);
+                            } else {
+                                return delegate().request(request);
+                            }
+                        }
+                    })
+                    .appendClientFilter(c -> new StreamingHttpClientFilter(c) {
+                        @Override
+                        protected Single<StreamingHttpResponse> request(final StreamingHttpRequester delegate,
+                                                                        final StreamingHttpRequest request) {
+                            if (filterType.equals(FilterType.CLIENT)) {
+                                return transformer.apply(delegate, request);
+                            } else {
+                                return delegate.request(request);
+                            }
+                        }
+                    })
+                    .buildStreaming()) {
+
+                int numRequests = 5;
+                for (int i = 0; i < numRequests; i++) {
+                    if (expectedException == null) {
+                        StreamingHttpResponse response = client.request(client.get("/")).toFuture().get();
+                        assertEquals(HttpResponseStatus.OK, response.status());
+                        // Consume the body to release the connection back to the pool
+                        response.messageBody().ignoreElements().toFuture().get();
+                    } else {
+                        ExecutionException ex = assertThrows(ExecutionException.class,
+                                () -> client.request(client.get("/")).timeout(Duration.ofMillis(100)).toFuture().get());
```

Comment:
This is weird, but definitely not related to changes in this PR. Consider either skipping tests that target exceptions originated by connection level filter or keeping this timeout. We can do a follow-up PR to investigate the root cause.

Reply:
I'll skip the failing ones since it makes it a little awkward to test the assertions. I think in the follow-up PR we should try to re-add them back if we figure out the root cause.